README.txt
Hermod

-Short intro
This is the program and resource files for running automagical inversions at
RSS.

hermod/files-nilu
find and retrieve new files from zardoz.nilu.no to make ZPT-files.

hermod/files-pdc
find and retrieve new files from pdc.kth.se.

hermod/l1b
Filehandling and processing before jobs are sent to the queue

hermod/l2
Filehandleng and processing after inversion are done.


ChangeLog
2007-02-13 Reloase 1.2.2 <joakim.moller@chalmers.se>

2006-12-20 Release 1.1.0 <joakim.moller@chalmers.se>
	* scrips/readFreq: obsolete, startscript in Level1b take care of
	adding into dadabase.
	* l2/level2.py,l1b/level1b.py,l1b/transitions.py: Adaptions to make
	hermod more failsafe.

2006-12-13 Get PDCfiles <joakim.moller@chalmers.se>
	* scripts/getL1b: Downloads files AC1/AC2 or both from PDC. Reads list
	from stin

2006-11-17 Back fixes <joakim.moller@chalmers.se>
	* l2/transitions.py: Adapted to new queue configuration.
	* l2/level2.py: Adds jobs to queue with sql keyword REVERSED
	* pdc/pdcTools.py: Add tools to get files from PDC
	* scripts/getL1bFiles: Downloads a range of files from PDC
	
2006-10-02 Testing plotting/filecheck <joakim.moller@chalmers.se>
	* l2/smrlevel2.py: classes for quicklook and verification of database
	added.

2006-09-25 Release 1.0.2 <joakim.moller@chalmers.se>
	* scripts/rerunOrbits: Setting fqid=0 reruns all active modes.

2006-08-16 Release 1.0.1 <joakim.moller@chalmers.se>
	* l1b/rerunOrbits.py, l1b/insertFiles.py, l2/readFreq.py: Removed, not
	used any more.

	* l1b/level1b.py: Creates directories in the right way.

2006-08-14 Release 1.0.0rc1 <joakim.moller@chalmers.se>

	* l1b/levele1b.py, l1b/transitions.py, l2/level2.py,
	scripts/rerunOrbits, scripts/readFreq, scripts/insertFiles, setup.py:
	Are members of the 1.0.0rc1.

2006-08-13 Bugfixes of last change <joakim.moller@chalmers.se>

	* l1b/level1b.py, l1b/transitions. py,l2/level2.py: Fixed typos,
	logical errors, referenses etc.

	* scripts/insertFiles, scripts/readFreq, scripts/rerunOrbits: fixed
	errors.

2006-08-10 Classes and scripts - Hermod package <joakim.moller@chalmers.se>

	* l2/level2.py, l1b/level1b.py, l1b/transitions.py: Classes providing
	functions to the scripts in scrips/.
	
	* l1b/insertfiles.py, l2/readFreq.py, l2/rerunOrbits.py: Main methods
	are moved to scripts.

	* scripts/insertfile.py,scripts/rerunOrbits.py,scripts/readFreq.py:
	Are scripts using the hermod package. Same funktinality provided by
	the scripts but they are using the hermed package

2006-08-09 Bugfixes of last change <joakim.moller@chalmers.se>

	* l1b/insertFiles.py,l1b/transitions.py: a lot of small bugs
	corrected.

2006-08-08 Objectorieted insertFiles<joakim.moller@chalmers.se>

	* l1b/insertFiles.py: Total remake of structural design.

	* l1b/transitions.py: Object that handles SMR-structure and submitting
	jobs to queue.

2006-08-02 Objectoriented readFreq <joakim.moller@chalmers.se>

	* l2/readFreq.py: Total remake of structural design - towards making
	it easy to reuse code or use it in the python interpreter.

	* l2/collectStd.py: removed, not needed anymore

2006-08-01 Major changes <joakim.moller@chalmers.se>
	
	* Almost all files: fqid is a new parameter in level2 table. Major
	parts of the sourcecode was rewritten. No fundamental changes in
	functionality

	* l1b/insertFiles.py: a compiled matlab script runs instead of
	create_tp

	* l1b/rerunOrbits.py: changed in-parameter. see usage()

	
2005-10-19 Error to stderr <joakim.moller@chalmers.se>

	* l1b/insertFiles.py: Moving files to a spool_failure director i files not
	having more than 3 scan. Also errors are written to stderr.
	
2005-10-06 Bad orbit files <joakim.moller@chalmers.se>

	* l1b/insertFiles.py: Added a check to find bad orbit files, files
	with less than 3 scans are ignored.

	* l1b/insert_level1b.awk, l2/test, l2/update.log,
	l2/update_l2_files.awk, l2/update_level2_files.awk: removed, not used anymore.
2005-09-19 Freqmode search <joakim.moller@chalmers.se>

	* l2/readFreq.py: Problems to read 3digit hexorbit number fixed

	* l1b/rerunOrbits.py: Fixed and operational, but not safe.

	* l1b/insertFiles.py: Fixed problem with 3digits.

2005-09-19 Freqmode search <joakim.moller@chalmers.se>

	* l1b/rerunOrbit.py: Added a inparameter to make it easier to search
	for not processed data.

2005-09-19 Three digit bugfix <joakim.moller@chalmers.se>

	* l1b/readFreq.py: Adds zeros in front of the orbitnumber if it has
	less than 4 digits. 

2005-09-16 Some bugfixes and cleanup <joakim.moller@chalmers.se>

	* l1b/read_hdf.c: Compilation instructions in file

	* l1b/inserFiles.py: Some very verbose printouts removed

	* l1b/msl.awk, l1b/freqmodes, l1b/freqmodes_nr, l1b/freqmodes_str:
	removed.

	* l1b/rerunOrbits.py: Tool to rerun orbit files. Note this is program
	is not tested yet. Use were carefully! Talk to Joakim first.

2005-08-29 Pythonised startscript <joakim.moller@chalmers.se>

	* l1b/insertFiles.py: Problems with different kinds of calibration
	solved. Now '3' is the one and only calibration spectrum.

2005-08-25 Pythonised startscript <joakim.moller@chalmers.se>
	
	* l1b/insertFiles.py: Inserts all files in spooldir into odin
	database, reorganises files according to SMR-directory structure.

2005-07-18 Joakim Möller <joakim.moller@chalmers.se>
	
	* l1b/msl.awk: Changing directory to ~logs/ before sending job to
	queue. This to avoid logfiles in odinops home directory.

2005-07-15 Hermod in cvs <joakim.moller@chalmers.se>

	* files-pdc/: Scripts that fetches files from pdc

	* files-nilu/: Scripts that fetches files from nilu

	* l1b/: gawk-scripts, adds l1b to the database and starts processings

	* l2b/: python-scripts, adds l2-files to the database.


mat/poff.m
function poff(filename)

format long g;
a=load(filename,'Xs','L','SMR');
fprintf('\n');
%find pointing offset cell

for i=[1:size(a.Xs,2)], 
    if not (isempty(a.Xs{i})),
        point =1;
        for j=[1:size(a.Xs{i},1)],
            if strcmp(a.Xs{i}{j}.name,'Pointing: off-set'),
                point = j;
            end
        end
        poff=a.Xs{i}{point}.x;
        fprintf('!!,%d,%0.20d\n',i,poff)
        %sprintf('update low_priority ignore level2,scans set poffset=%f where calibration=%d and scan=%d and hex(orbit)=''%X'' and freqmode=%d and level2.id=scans.id;\n',a.Xs{i}{point}.x,bitand(a.L.LEVEL,255),i,a.L.ORBIT,strnr(a.SMR.FREQMODE));
    end;
end;
return

function x=strnr(str)
switch str 
case 'SM_AC2a'
    nr = 1;
case 'SM_AC2b' 
    nr= 1;
case 'SM_AC2ab' 
    nr= 1;
case 'SM_AC1e'
    nr= 2;
case 'HM_AC1f' 
    nr= 24;
case 'IM_AC2a' 
    nr= 8;
case 'IM_AC2b' 
    nr= 8;
case 'IM_AC2ab' 
    nr= 8;
case 'IM_AC2c' 
    nr= 17;
case 'IM_AC1c' 
    nr= 19;
case 'IM_AC1de' 
    nr= 21;
case 'HM_AC1c' 
    nr= 13;
case 'HM_AC2c' 
    nr= 14;
case 'HM_AC2ab' 
    nr= 22;
case 'HM_AC1d' 
    nr= 23;
case 'HM_AC1e' 
    nr= 23;
case 'TM_ACs1' 
    nr= 25;
case 'TM_AOS1' 
    nr= 26;
case 'TM_ACs2' 
    nr= 27;
case 'TM_AOS2' 
    nr= 28;
case 'SM_FB' 
    nr= 20;
case 'HM_AC1a' 
    nr= 3;
case 'HM_AC1b' 
    nr= 3;
case 'HM_AC2a' 
    nr= 4;
case 'HM_AC2b' 
    nr= 4;
case 'HM_AOS2' 
    nr= 5;
case 'NM_AC1a' 
    nr= 6;
case 'NM_AC1b' 
    nr= 6;
case 'NM_AC2a' 
    nr= 7;
case 'NM_AC2b' 
    nr= 7;
case 'IM_AC1a' 
    nr= 9;
case 'IM_AC1b' 
    nr= 9;
case 'SM_AC2c' 
    nr= 10;
case 'SM_AC2d' 
    nr= 10;
case 'HM_AOS1' 
    nr= 11;
case 'IM_AOS1' 
    nr= 12;
case 'HM_AC2d' 
    nr= 15;
case 'HM_AC2e' 
    nr= 15;
case 'HM_AC2d' 
    nr= 16;
case 'HM_AC2f' 
    nr= 16;
otherwise 
    nr=0;
end
x=nr;
